js/src/util/index.js

```diff
@@ -252,6 +234,35 @@ const execute = callback => {
   }
 }
 
+const executeAfterTransition = (callback, transitionElement, waitForTransition = true) => {
+  if (!waitForTransition) {
+    execute(callback)
+    return
+  }
+
+  const durationPadding = 5
+  const emulatedDuration = getTransitionDurationFromElement(transitionElement) + durationPadding
+
+  let called = false
```

Comment:
This flag variable (`called`) can not be omitted? @alpadev :thinking:

Reply:
Although, there is no test covering the use of the `called` flag :slightly_smiling_face:

Reply:
Not sure about omitting this. I mean in theory we could, since the event handler unregisters itself, thus the callback is only executed once. But there would be another `transitionend` event dispatched by the `setTimeout` which could be a problem.

`called` is partly covered by
https://github.com/twbs/bootstrap/blob/fd016ac7ba9900f2275163d86c0d554d6a0d9b5a/js/tests/unit/util/index.spec.js#L665 because `setTimeout` wouldn't trigger an event if `called` is true. But I added another test that should satisfy your objection. (I can't really verify the state of called since it's scoped to the function.)